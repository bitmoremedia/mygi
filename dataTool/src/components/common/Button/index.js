import React, { PropTypes } from 'react'

import { Button, DangerButton } from './styles'

const ButtonComponent = ({ children, onClick, danger }) => {
  if (danger) {
    return (
      <DangerButton onClick={onClick} >
        {children}
      </DangerButton>
    )
  }
  return (
    <Button onClick={onClick} >
      {children}
    </Button>
  )
}

ButtonComponent.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func,
  danger: PropTypes.bool,
}

ButtonComponent.defaultProps = {
  onClick: undefined,
  danger: false,
}

export default ButtonComponent
