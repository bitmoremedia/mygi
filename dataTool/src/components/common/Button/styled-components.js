import styled from 'styled-components'

export const Button = styled.button`
  background-color: black;
  color: white;
  cursor: pointer;
  border-radius: 1px;
  border: 0px;
  padding: 5px 10px 5px 10px;
  font-size: 14px;
  &:focus {
      outline:none;
  }
`

export const DangerButton = styled(Button)`
  background-color: red;
`
