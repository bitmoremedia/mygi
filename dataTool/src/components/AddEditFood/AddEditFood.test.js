import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import AddEditFoodComponent from './component'
import Button from '../common/Button'

test('component renders without error', () => {
  const wrapper = mount(
    <AddEditFoodComponent addFoodItem={() => {}} deleteFoodItem={() => {}} />,
  )
  expect(wrapper.exists()).toBe(true)
})

// INTERACTIVITY TESTS

test('can add food item ONLY if valid', () => {
  const addFoodItem = jest.fn()
  const wrapper = mount(
    <AddEditFoodComponent addFoodItem={addFoodItem} deleteFoodItem={() => {}} />,
  )
  // attempt to submit without adding the required data
  wrapper.find('form').simulate('submit')
  expect(addFoodItem).not.toHaveBeenCalled()
  // add required data and attempt to submit again
  wrapper.find('input[name="foodName"]').simulate('change', { target: { name: 'foodName', value: 'Kidney Bean' } })
  wrapper.find('input[name="giValue"]').simulate('change', { target: { name: 'giValue', value: 35 } })
  wrapper.find('form').simulate('submit')
  expect(addFoodItem).toHaveBeenCalled()
})

test('can delete food item', () => {
  const deleteFoodItem = jest.fn()
  const foodItem = {
    name: 'Kidney Beans',
    id: 'some-unique-id',
    gi: 33,
    sources: {},
    category: 'Beans',
  }
  const wrapper = mount(
    <AddEditFoodComponent
      addFoodItem={() => {}}
      mode="edit"
      foodItem={foodItem}
      deleteFoodItem={deleteFoodItem}
    />,
  )
  wrapper.find(Button).last().simulate('click')
  expect(deleteFoodItem).toHaveBeenCalledWith({ foodId: 'some-unique-id' })
})

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const component = renderer.create(
    <AddEditFoodComponent addFoodItem={() => {}} deleteFoodItem={() => {}} />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('component renders correctly in edit mode', () => {
  const foodItem = {
    name: 'Kidney Beans',
    id: 'some-unique-id',
    gi: 33,
    sources: {},
    category: 'Beans',
  }
  const component = renderer.create(
    <AddEditFoodComponent
      addFoodItem={() => {}}
      deleteFoodItem={() => {}}
      mode="edit"
      foodItem={foodItem}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('component renders correctly in add mode with initial values provided', () => {
  const initialValues = {
    name: 'Kidney Beans',
    gi: 33,
  }
  const component = renderer.create(
    <AddEditFoodComponent
      addFoodItem={() => {}}
      deleteFoodItem={() => {}}
      mode="add"
      initialValues={initialValues}
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
