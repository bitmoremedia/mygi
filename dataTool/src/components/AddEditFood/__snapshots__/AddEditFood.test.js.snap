exports[`test component renders as expected 1`] = `
<div
  className="cEMVUV">
  <form
    onSubmit={[Function]}>
    <div
      className="jhQCSn">
      <div
        className="egpniL">
        <input
          autoComplete="off"
          className="jeXhuP"
          name="foodName"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value="" />
      </div>
      <div
        className="egpniL">
        <select
          className="XDeHG"
          name="category"
          onChange={[Function]}
          value="">
          <option
            defaultValue={true}
            disabled={true}
            value="">
            Select Category
          </option>
          <option
            value="Beans">
            Beans
          </option>
          <option
            value="Breads">
            Breads
          </option>
          <option
            value="Cereals">
            Cereals
          </option>
          <option
            value="Dairy">
            Dairy
          </option>
          <option
            value="Fruits">
            Fruits
          </option>
          <option
            value="Snacks & Sweets">
            Snacks & Sweets
          </option>
          <option
            value="Staples">
            Staples
          </option>
          <option
            value="Vegetables">
            Vegetables
          </option>
        </select>
      </div>
      <div
        className="egpniL">
        <input
          autoComplete="off"
          className="jeXhuP"
          name="giValue"
          onChange={[Function]}
          placeholder="GI Value"
          type="number"
          value="" />
      </div>
      <div
        className="egpniL">
        <div
          className="gcxdPw">
          <button
            className="OgMJe"
            onClick={undefined}>
            Add
          </button>
        </div>
      </div>
      <div
        className="egpniL">
        <div
          className="dEIiuk">
          
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`test component renders correctly in add mode with initial values provided 1`] = `
<div
  className="cEMVUV">
  <form
    onSubmit={[Function]}>
    <div
      className="jhQCSn">
      <div
        className="egpniL">
        <input
          autoComplete="off"
          className="jeXhuP"
          name="foodName"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value="Kidney Beans" />
      </div>
      <div
        className="egpniL">
        <select
          className="XDeHG"
          name="category"
          onChange={[Function]}
          value="">
          <option
            defaultValue={true}
            disabled={true}
            value="">
            Select Category
          </option>
          <option
            value="Beans">
            Beans
          </option>
          <option
            value="Breads">
            Breads
          </option>
          <option
            value="Cereals">
            Cereals
          </option>
          <option
            value="Dairy">
            Dairy
          </option>
          <option
            value="Fruits">
            Fruits
          </option>
          <option
            value="Snacks & Sweets">
            Snacks & Sweets
          </option>
          <option
            value="Staples">
            Staples
          </option>
          <option
            value="Vegetables">
            Vegetables
          </option>
        </select>
      </div>
      <div
        className="egpniL">
        <input
          autoComplete="off"
          className="jeXhuP"
          name="giValue"
          onChange={[Function]}
          placeholder="GI Value"
          type="number"
          value={33} />
      </div>
      <div
        className="egpniL">
        <div
          className="gcxdPw">
          <button
            className="OgMJe"
            onClick={undefined}>
            Add
          </button>
        </div>
      </div>
      <div
        className="egpniL">
        <div
          className="dEIiuk">
          
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`test component renders correctly in edit mode 1`] = `
<div
  className="cEMVUV">
  <h3>
    Edit: 
    Kidney Beans
  </h3>
  <form
    onSubmit={[Function]}>
    <div
      className="jhQCSn">
      <div
        className="egpniL">
        <input
          autoComplete="off"
          className="jeXhuP"
          name="foodName"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value="Kidney Beans" />
      </div>
      <div
        className="egpniL">
        <select
          className="XDeHG"
          name="category"
          onChange={[Function]}
          value="Beans">
          <option
            defaultValue={true}
            disabled={true}
            value="">
            Select Category
          </option>
          <option
            value="Beans">
            Beans
          </option>
          <option
            value="Breads">
            Breads
          </option>
          <option
            value="Cereals">
            Cereals
          </option>
          <option
            value="Dairy">
            Dairy
          </option>
          <option
            value="Fruits">
            Fruits
          </option>
          <option
            value="Snacks & Sweets">
            Snacks & Sweets
          </option>
          <option
            value="Staples">
            Staples
          </option>
          <option
            value="Vegetables">
            Vegetables
          </option>
        </select>
      </div>
      <div
        className="egpniL">
        <input
          autoComplete="off"
          className="jeXhuP"
          name="giValue"
          onChange={[Function]}
          placeholder="GI Value"
          type="number"
          value={33} />
      </div>
      <div
        className="egpniL">
        <div
          className="gcxdPw">
          <button
            className="OgMJe"
            onClick={undefined}>
            Save
          </button>
        </div>
      </div>
      <div
        className="egpniL">
        <div
          className="dEIiuk">
          
        </div>
      </div>
    </div>
  </form>
  <div
    className="gcxdPw">
    <button
      className="feUtjL"
      onClick={[Function]}>
      Delete
    </button>
  </div>
</div>
`;
